src/domain/community/user/user.service.authorization.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { AuthorizationCredential, AuthorizationPrivilege } from '@common/enums';
+import { User, IUser } from '@domain/community/user';
+import { AgentService } from '@domain/agent/agent/agent.service';
+import { AuthorizationRule } from '@src/services/authorization-engine/authorizationRule';
+import { AuthorizationEngineService } from '@src/services/authorization-engine/authorization-engine.service';
+import { UserService } from './user.service';
+import { ProfileAuthorizationService } from '../profile/profile.service.authorization';
+
+@Injectable()
+export class UserAuthorizationService {
+  constructor(
+    private authorizationEngine: AuthorizationEngineService,
+    private profileAuthorizationService: ProfileAuthorizationService,
+    private agentService: AgentService,
+    private userService: UserService,
+    @InjectRepository(User)
+    private userRepository: Repository<User>
+  ) {}
+
+  async applyAuthorizationRules(user: IUser): Promise<IUser> {
+    user.authorizationRules = this.createAuthorizationRules(user.id);
+
+    // cascade
+    const profile = this.userService.getProfile(user);
```

Comment:
for discussion: cascading looks like a lot of copy / pasted code - here base functionality / interface would make sense.